packages/debug/src/node/debug-adapter-session.ts

```diff
@@ -34,25 +34,23 @@ import { DisposableCollection, Disposable } from '@theia/core/lib/common/disposa
  */
 export class DebugAdapterSessionImpl implements DebugAdapterSession {
 
-    private static TWO_CRLF = '\r\n\r\n';
-    private static CONTENT_LENGTH = 'Content-Length';
-
     private readonly toDispose = new DisposableCollection();
     private channel: IWebSocket | undefined;
-    private contentLength: number;
-    private buffer: Buffer;
 
     constructor(
         readonly id: string,
         protected readonly communicationProvider: CommunicationProvider
     ) {
-        this.contentLength = -1;
-        this.buffer = Buffer.alloc(0);
         this.toDispose.pushAll([
             this.communicationProvider,
             Disposable.create(() => this.write(JSON.stringify({ seq: -1, type: 'request', command: 'disconnect' }))),
             Disposable.create(() => this.write(JSON.stringify({ seq: -1, type: 'request', command: 'terminate' })))
         ]);
+
+        this.communicationProvider.onMessageReceived((message: string) => this.send(message));
```

Comment:
Nitpick:
```suggestion
        this.communicationProvider.onMessageReceived(message => this.send(message));
```

Reply:
I kinda like to repeat the types when they are not obvious from the context. And in this code, we're stringifying/parsing messages all the time, so it matters to know the types.

Reply:
Sure, alright with me 👍